drivers/SmartThings/sonos/src/disco.lua

```diff
@@ -37,18 +37,17 @@ end
 
 function Discovery.discover(driver, _, should_continue)
   log.info("Starting Sonos discovery")
-  local known_devices_dnis = {}
-  local found_ip_addrs = {}
+  driver.found_ips = {}
 
-  local device_list = driver:get_devices()
-  for _, device in ipairs(device_list) do
-    local id = device.device_network_id
-    known_devices_dnis[id] = true
-  end
+  while should_continue() do
+    local known_devices_dnis = {}
 
-  driver.found_ips = found_ip_addrs
+    local device_list = driver:get_devices()
+    for _, device in ipairs(device_list) do
+      local id = device.device_network_id
+      known_devices_dnis[id] = true
+    end
```

Comment:
would it be worthwhile to have `know_devices_dnis` declared outside this loop so it doesn't get re-built every time, or is that intentional?

Reply:
I intentionally moved it inside the loop; I had the same thought originally which is why it was outside the loop but if you think about it, a device can join while discovery is still happening meaning that the `known_dnis` won't update and you'll re-scan that device.

And for the sake of consistency and correctness I don't want to populate that `known_dnis` table with DNIs from anywhere other than the driver's canonical list of known DNI's, so I don't want to update them from the discovery code manually.